src/main/java/org/folio/linked/data/validation/dto/LccnPatternValidator.java

```diff
@@ -0,0 +1,36 @@
+package org.folio.linked.data.validation.dto;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.regex.Pattern;
+import org.folio.linked.data.domain.dto.LccnRequest;
+import org.folio.linked.data.validation.LccnPatternConstraint;
+
+public class LccnPatternValidator implements ConstraintValidator<LccnPatternConstraint, LccnRequest> {
+
+  private static final Pattern LCCN_STRUCTURE_A_PATTERN = Pattern.compile("( {3}|[a-z][ |a-z]{2})\\d{8} ");
+  private static final Pattern LCCN_STRUCTURE_B_PATTERN = Pattern.compile("( {2}|[a-z][ |a-z])\\d{10}");
+
+  @Override
+  public boolean isValid(LccnRequest lccnRequest, ConstraintValidatorContext constraintValidatorContext) {
+    if (isCurrent(lccnRequest)) {
+      return lccnRequest.getValue()
+        .stream()
+        .noneMatch(this::hasInvalidPattern);
+    } else {
+      return true;
+    }
+  }
+
+  private boolean isCurrent(LccnRequest lccnRequest) {
+    return lccnRequest.getStatus()
```

Comment:
Hi @askhat-abishev,
When user do not choose any status in the UI (refer image), then also we have to consider the LCCN as current.

So, we will have to change this logic as follows

```
isEmpty(lccnRequest.getStatus()) || ... the current logic
```

<img width="676" alt="Screenshot 2024-11-18 at 11 31 12 AM" src="https://github.com/user-attachments/assets/c36bd8bd-494d-4ae6-a8c5-e67a7370ab64">

Reply:
done